pkg/container/docker/images.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+// ImageID returns the image ID for a given image[:tag]
+func ImageID(image string) (string, error) {
+	cmd := exec.Command("docker", "images", "-q", image)
+	lines, err := exec.CombinedOutputLines(cmd)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get the id of name %q", image)
+	}
+	if len(lines) != 1 {
+		return "", errors.Errorf("failed to get the id of name %q", image)
```

Comment:
if we're going to return an error for this case it should be a well known / exporter error and documented so that callers can check for this not-surprising case.

alternatively, `"", nil` could signal there isn't one, since it hsouldn't be possible to have an image without an id?

Reply:
you can pass a  non-existent image name, I think we need to fail in that case.

```
./kind load docker-image nonexistanteiamge
Error: There is no image id with name "nonexistanteiamge:latest"
```

what do you mean by this? 

> well known / exporter error and documented